presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -303,10 +304,17 @@ public Expression getJoinCriteria(Join join)
     public void recordSubqueries(Node node, ExpressionAnalysis expressionAnalysis)
     {
         NodeRef<Node> key = NodeRef.of(node);
-        this.inPredicatesSubqueries.putAll(key, expressionAnalysis.getSubqueryInPredicates());
-        this.scalarSubqueries.putAll(key, expressionAnalysis.getScalarSubqueries());
-        this.existsSubqueries.putAll(key, expressionAnalysis.getExistsSubqueries());
-        this.quantifiedComparisonSubqueries.putAll(key, expressionAnalysis.getQuantifiedComparisons());
+        this.inPredicatesSubqueries.putAll(key, dereference(expressionAnalysis.getSubqueryInPredicates()));
+        this.scalarSubqueries.putAll(key, dereference(expressionAnalysis.getScalarSubqueries()));
+        this.existsSubqueries.putAll(key, dereference(expressionAnalysis.getExistsSubqueries()));
+        this.quantifiedComparisonSubqueries.putAll(key, dereference(expressionAnalysis.getQuantifiedComparisons()));
+    }
+
+    private <T extends Node> List<T> dereference(Collection<NodeRef<T>> nodeRefs)
```

Comment:
I see nothing non-general-purpose in its body or signature, and I expect to see more of its copies throughout the codebase soon enough to risk adding an (IMO) generic method only used once to the `NodeRef`. Not insisting though :)